astropy/timeseries/periodograms/lombscargle/implementations/utils.py

```diff
@@ -79,7 +87,213 @@ def extirpolate(x, y, N=None, M=4):
     return result
 
 
-def trig_sum(t, h, df, N, f0=0, freq_factor=1, oversampling=5, use_fft=True, Mfft=4):
+if HAS_SCIPY:
```

Comment:
If you want to define APIs that are only available if scipy is installed, I think it would make more sense to put them into a dedicated module instead of conditionally skipping their definition.